src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php

```diff
@@ -0,0 +1,477 @@
+    ) {
+        $this->table = $table;
+        $this->dynamoDb = $dynamo;
+        $this->keyAttribute = $keyAttribute;
+        $this->valueAttribute = $valueAttribute;
+        $this->expirationAttribute = $expirationAttribute;
+
+        $this->setPrefix($prefix);
+    }
+
+    /**
+     * Retrieve an item from the cache by key.
+     *
+     * @param string|array $key
+     *
+     * @return mixed
+     */
+    public function get($key)
+    {
+        if (\is_array($key)) {
+            return $this->many($key);
+        }
+
+        $response = $this->dynamoDb->getItem([
+            'TableName' => $this->table,
+            'ConsistentRead' => false,
+            'Key' => [
+                $this->keyAttribute => [
+                    'S' => $this->prefix . $key,
+                ],
+            ],
+        ]);
+
+        $item = $response->getItem();
+        if (empty($item)) {
+            return;
+        }
+
+        if ($this->isExpired($item)) {
+            return;
+        }
+
+        if (isset($item[$this->valueAttribute])) {
+            return $this->unserialize(
+                $item[$this->valueAttribute]->getS() ??
+                $item[$this->valueAttribute]->getN() ??
+                null
+            );
+        }
+    }
+
+    /**
+     * Retrieve multiple items from the cache by key.
+     *
+     * Items not found in the cache will have a null value.
+     *
+     * @return array
+     */
+    public function many(array $keys)
+    {
+        $result = [];
+        foreach ($keys as $key) {
+            $result[$key] = $this->get($key);
+        }
+
+        // TODO Use BatchWriteItem. Blocked by https://github.com/async-aws/aws/issues/566
+        return $result;
+    }
+
+    /**
+     * Store an item in the cache for a given number of seconds.
+     *
+     * @param string $key
+     * @param mixed  $value
+     * @param int    $seconds
+     *
+     * @return bool
+     */
+    public function put($key, $value, $seconds)
+    {
+        $this->dynamoDb->putItem([
+            'TableName' => $this->table,
+            'Item' => [
+                $this->keyAttribute => [
+                    'S' => $this->prefix . $key,
+                ],
+                $this->valueAttribute => [
+                    $this->type($value) => $this->serialize($value),
+                ],
+                $this->expirationAttribute => [
+                    'N' => (string) $this->toTimestamp($seconds),
+                ],
+            ],
+        ]);
+
+        return true;
+    }
+
+    /**
+     * Store multiple items in the cache for a given number of $seconds.
+     *
+     * @param int $seconds
+     *
+     * @return bool
+     */
+    public function putMany(array $values, $seconds)
+    {
+        foreach ($values as $key => $value) {
+            $this->put($key, $value, $seconds);
+        }
+
+        // TODO Use BatchWriteItem. Blocked by https://github.com/async-aws/aws/issues/566
+        return true;
```

Comment:
We should use BatchWriteItem here instead. Related to https://github.com/async-aws/aws/issues/566